tests/regression_tests/unstructured_mesh/source_sampling/test.py

```diff
@@ -0,0 +1,190 @@
+    water_mat.set_density("atom/b-cm", 0.07416)
+    materials.append(water_mat)
+
+    materials.export_to_xml()
+
+    ### Geometry ###
+    dimen = 10
+    size_hex = 20.0/dimen
+
+    ### Geometry ###
+    cell = np.empty((dimen, dimen, dimen), dtype=object)
+    surfaces = np.empty((dimen+1, 3), dtype=object)
+
+    geometry = openmc.Geometry()
+    universe = openmc.Universe(universe_id=1, name="Contains all hexes")
+
+    for i in range(0,dimen+1):
+        surfaces[i][0] = openmc.XPlane(-10.0+i*size_hex, name="X plane at "+str(-10.0+i*size_hex))
+        surfaces[i][1] = openmc.YPlane(-10.0+i*size_hex, name="Y plane at "+str(-10.0+i*size_hex))
+        surfaces[i][2] = openmc.ZPlane(-10.0+i*size_hex, name="Z plane at "+str(-10.0+i*size_hex))
+
+        surfaces[i][0].boundary_type = 'vacuum'
+        surfaces[i][1].boundary_type = 'vacuum'
+        surfaces[i][2].boundary_type = 'vacuum'
+
+    for k in range(0,dimen):
+        for j in range(0,dimen):
+            for i in range(0,dimen):
+                cell[i][j][k] = openmc.Cell(name=("x " + str(i) +" y " + str(j) +" z " + str(k)))
+                cell[i][j][k].region = +surfaces[i][0] & -surfaces[i+1][0] & \
+                                    +surfaces[j][1] & -surfaces[j+1][1] & \
+                                    +surfaces[k][2] & -surfaces[k+1][2]
+                cell[i][j][k].fill = water_mat
+                universe.add_cell(cell[i][j][k])
+
+    geometry = openmc.Geometry(universe)
+
+    mesh_filename = "test_mesh_tets.e"
+
+    uscd_mesh = openmc.UnstructuredMesh(mesh_filename, test_opts['library'])
+
+    ### Tallies ###
+
+    # create tallies
+    tallies = openmc.Tallies()
+
+    tally1 = openmc.Tally(1)
+    tally1.scores = ['scatter', 'total', 'absorption']
+    # Export tallies
+    tallies = openmc.Tallies([tally1])
+    tallies.export_to_xml()
+
+    ### Settings ###
+    settings = openmc.Settings()
+    settings.run_mode = 'fixed source'
+    settings.particles = 5000
+    settings.batches = 2
+
+    # settings.create_fission_neutrons = False
+    settings.tracks = [(1, 1, 1)]
+    settings.max_tracks = 10000
+
+    # source setup
+    if test_opts['schemes'] == 'volume':
+        space = openmc.stats.MeshIndependent(volume_weighting=True, mesh=uscd_mesh)
+    elif test_opts['schemes'] == 'file':
+        array = np.zeros(12000)
+        for i in range(0, 12):
+            array[i] = 10
+            array[i+12] = 2
+        space = openmc.stats.MeshIndependent(volume_weighting=False, weights_from_file=array, mesh=uscd_mesh)
+
+    energy = openmc.stats.Discrete(x=[15.e+06], p=[1.0])
+    source = openmc.Source(space=space, energy=energy)
+    settings.source = source
+
+    model = openmc.model.Model(geometry=geometry,
+                               materials=materials,
+                               tallies=tallies,
+                               settings=settings)
+
+    harness = UnstructuredMeshSourceTest('statepoint.2.h5',

```

Comment:
Indentation fixes needed here

Reply:
Should they be lined up with the first variable set? So for instance the m from model should be in the same column as the ' from 'statepoint.2.h5'?

Reply:
Yep yep